fix(api,plugins): yaml-context AtlasMcpTool import + tsgo errors after #2190

Type-checking `packages/api` (tsgo --noEmit) on `main` (commit 43de507d) surfaces 5 errors stemming from `plugins/yaml-context/src/index.ts`:

```
src/lib/tools/__tests__/description-rubric.test.ts(174,34): error TS2339: Property 'mcpTools' does not exist on type 'AtlasContextPlugin<ContextYamlConfig>'.
../../plugins/yaml-context/src/index.ts(27,3): error TS2305: Module '"@useatlas/plugin-sdk"' has no exported member 'AtlasMcpTool'.
../../plugins/yaml-context/src/index.ts(286,21): error TS7031: Binding element 'filter' implicitly has an 'any' type.
../../plugins/yaml-context/src/index.ts(286,31): error TS7006: Parameter 'ctx' implicitly has an 'any' type.
../../plugins/yaml-context/src/index.ts(343,5): error TS2353: Object literal may only specify known properties, and 'mcpTools' does not exist in type 'AtlasContextPlugin<ContextYamlConfig>'.
```

Reproduces on a clean `main` checkout with `bun install && cd packages/api && bun x tsgo --noEmit` — likely a regression from #2190 (Plugin SDK MCP tools extension point) where `AtlasMcpTool` was renamed/moved from `@useatlas/plugin-sdk` and `mcpTools` was added to a different plugin variant than `AtlasContextPlugin`.

The reference impl in `plugins/yaml-context` declares `mcpTools` on `AtlasContextPlugin<ContextYamlConfig>` but the type doesn't carry that field — only `AtlasMcpTool`-shaped plugins do.

Fix likely: re-export `AtlasMcpTool` from the plugin-sdk barrel or move the `mcpTools` array to a separate plugin variant that the host can compose with `AtlasContextPlugin`.

Found while implementing #2073.

--- a/packages/plugin-sdk/package.json
+++ b/packages/plugin-sdk/package.json
@@ -6,7 +6,6 @@
   "scripts": {
     "build": "rm -rf dist && bun build src/index.ts src/types.ts src/helpers.ts src/ai.ts src/hono.ts src/testing.ts --outdir dist --target node --packages external && bun x tsc -p tsconfig.build.json",
     "prepare": "bun run build",
-    "prepublishOnly": "bun run build",
     "test": "bun test src/__tests__/types.test.ts && bun test src/__tests__/infer.test.ts && bun test src/__tests__/testing.test.ts"
   },
   "exports": {

--- a/packages/plugin-sdk/package.json
+++ b/packages/plugin-sdk/package.json
@@ -5,6 +5,7 @@
   "type": "module",
   "scripts": {
     "build": "rm -rf dist && bun build src/index.ts src/types.ts src/helpers.ts src/ai.ts src/hono.ts src/testing.ts --outdir dist --target node --packages external && bun x tsc -p tsconfig.build.json",
+    "prepare": "bun run build",
     "prepublishOnly": "bun run build",
     "test": "bun test src/__tests__/types.test.ts && bun test src/__tests__/infer.test.ts && bun test src/__tests__/testing.test.ts"
   },

--- a/packages/types/package.json
+++ b/packages/types/package.json
@@ -4,7 +4,8 @@
   "description": "Shared types for the Atlas text-to-SQL agent",
   "type": "module",
   "scripts": {
-    "build": "rm -rf dist && bun build src/index.ts src/auth.ts src/conversation.ts src/connection.ts src/action.ts src/scheduled-task.ts src/errors.ts src/semantic.ts src/share.ts src/billing.ts src/dashboard.ts src/mode.ts src/starter-prompt.ts src/integrations.ts src/email-provider.ts src/mcp.ts --outdir dist --root src --target node --packages external && bun x tsc -p tsconfig.build.json"
+    "build": "rm -rf dist && bun build src/index.ts src/auth.ts src/conversation.ts src/connection.ts src/action.ts src/scheduled-task.ts src/errors.ts src/semantic.ts src/share.ts src/billing.ts src/dashboard.ts src/mode.ts src/starter-prompt.ts src/integrations.ts src/email-provider.ts src/mcp.ts --outdir dist --root src --target node --packages external && bun x tsc -p tsconfig.build.json",
+    "prepare": "bun run build"
   },
   "exports": {
     ".": {

